make.js

```diff
@@ -1021,11 +1034,63 @@ CLI.gentask = function() {
             console.log(`Running \"npm install\" command in ${taskPath}`);
             run(`npm install`);
             cd(__dirname);
+
+            // copy package.json, package-lock.json and npm-shrinkwrap.json files from generated tasks to the Tasks\taskname\_buildConfig\nodeversion folder.
+            const fileNames = ['package.json', 'package-lock.json', 'npm-shrinkwrap.json'];
+            const buildConfigsPath = path.join(tasksPath, taskName, '_buildConfigs', configsString);
+            if (fs.existsSync(buildConfigsPath)) {
+                for (const fileName of fileNames) {
+                    const filePath = path.join(taskPath, fileName);
+                    if (fs.existsSync(filePath)) {
+                        console.log(`Copying ${fileName} from ${taskPath} to ${buildConfigsPath} folder.`);
+                        cp(filePath, path.join(buildConfigsPath, fileName));
+                    }
+                }
+            }
         }
     });
+
     fs.writeFileSync(makeOptionsPath, JSON.stringify(newMakeOptions, null, 4));
 }
 
+function generateTasksOnServerBuild() {
+    //generate tasks
+    taskList.forEach(function (taskName) {
+        //read configs from versionmap.txt
+        const versionMapPath = path.join(genTaskPath, taskName + ".versionmap.txt");
+        if (!fs.existsSync(versionMapPath)) {
+            console.error(`No ${taskName}versionmap.txt file found under _generated folder.`);
+        }
+        const configs = [];
+        var versionMapContents = fs.readFileSync(versionMapPath, { encoding: 'utf-8' });
+        const versionMapLines = versionMapContents.split('\n');
+        for (const line of versionMapLines) {
+            const config = line.split('|')[0].trim();
+            if (config != "Default") {
+                configs.push(config);
+            }
+        }
+        const configsArr = configs.join('|').slice(0, -1).replace(/-/g, '_');
```

Comment:
You can use [generateTasks](https://github.com/microsoft/azure-pipelines-tasks/blob/master/make-util.js#L1866) function to generate the task with passed parameters

Reply:
updated